tests/phpunit/tests/test-api.php

```diff
@@ -67,4 +79,92 @@ public function test_translated_objects_get_translation( $type ) {
 		$this->assertSame( 0, PLL()->model->$type->get_translation( $objects['en'], 'de' ) );
 		$this->assertSame( 0, PLL()->model->$type->get_translation( $objects['en'], 'chti' ) );
 	}
+
+	/**
+	 * @covers ::pll_insert_term
+	 *
+	 * @testWith ["category", true, true]
+	 *           ["category", false, true]
+	 *           ["post_tag", false, true]
+	 *           ["tr_custom_tax", false, true]
+	 *           ["category", false, false]
+	 *           ["category", true, false]
+	 *           ["post_tag", false, false]
+	 *           ["tr_custom_tax", false, false]
+	 *
+	 * @param string $taxonomy          Taxonomy name.
+	 * @param bool   $with_parent       Whether or not the term has a parent.
+	 * @param bool   $with_translations Whether or not the term has translations.
+	 * @return void
+	 */
+	public function test_pll_insert_term_happy_path( $taxonomy, $with_parent, $with_translations ) {
+		$languages    = array( 'en', 'fr', 'de' );
+		$translations = array();
+		foreach ( $languages as $i => $language ) {
+			$args            = array();
+			$is_default_lang = 0 === $i;
+
+			if ( $with_parent ) {
+				$parent = self::factory()->term->create_and_get(
+					array(
+						'name'     => $is_default_lang ? 'Foo' : "Foo {$language}",
+						'taxonomy' => $taxonomy,
+						'lang'     => $language,
+					)
+				);
+				$args['parent'] = $parent->term_id;
+			}
+
+			if ( $with_translations && ! empty( $translations ) ) {
+				$args['translations'] = $translations;
+			}
+
+			$result = pll_insert_term( 'Foo', $taxonomy, $language, $args );
+
+			$this->assertIsArray( $result, "The term in {$this->pll_env->model->get_language( $language )->name} could not be created." );
+			$this->assertArrayHasKey( 'term_id', $result );
+			$this->assertArrayHasKey( 'term_taxonomy_id', $result );
+
+			$translations[ $language ] = $result['term_id'];
+			$term                      = get_term( $result['term_id'], $taxonomy );
+
+			/*
+			 * @FIXME Polylang doesn't reuse parent slug and add a suffix with language slug instead...
+			 */
```

Comment:
You can remove the comment and open an issue instead.

Reply:
👍 
The [issue](https://github.com/polylang/polylang-pro/issues/2193).